File: 024-Turn All Bulbs On with Faulty Switches/TurnAllBulbsOnwithFaultySwitches.py
def min_switch_presses(A):
    toggle = 0
    count = 0
    
    for i in range(len(A)):
        current_state = A[i] ^ toggle
        if current_state == 0:
            count += 1
            toggle ^= 1
    return count


A = [0, 1, 0, 1]
print("Minimum switches:", min_switch_presses(A))

File: 024-Turn All Bulbs On with Faulty Switches/TurnAllBulbsOnwithFaultySwitches.cpp
#include<iostream>
#include<vector>
using namespace std;

int minSwitchPresses(vector<int>&A){
    int toggle = 0;
    int Count = 0;

    for(int i = 0; i < A.size(); i++){
        int CurrentState = A[i] ^ toggle;
        if(CurrentState == 0){
            Count++;
            toggle ^= 1;
        }
    }
    return Count;
}

int main(){
    vector<int> A = {0, 1, 0, 1};
    cout<<"Minimum Switches: "<<minSwitchPresses(A)<<endl;
    return 0;
}